### ISSUE TITLE
[220110]Neural Object Learning for 6D pose estimation using a Few cluttered Images

### ISSUE THREAD

**Issue by ChaeChae0505** [ChaeChae0505/Pytorch#4]
- 논문 : Neural Object Learning for 6D pose estimation using a Few cluttered Images
- 분류 : 6dof pose estimation
- 저자 : Kiru Park, Timothy Patten, and Markus Vincze
- 읽는 배경 : 6 dof pose estimation ycb-video dataset 이든 linemode든 접근이 좀 어려워서 가벼운 데이터를 찾다가 알게 되었다.
- 느낀점 : Transformer의 기초를 읽어보자 attention all ~
- 공부 참고자료
---

## 요약
- 3D model에 대한 texture 정보가 필요한데 우리는 그거 없이도 사용가능

## Introduction
- Mask r-cnn이나 , Deep residual learning image recognition과 같은 Image의 feature를 학습
시키는 방법들이 존재하는데 이걸 포함하는 pose estimation method들은  [21,24,27,31,34,35,45]. 등이 있다. 이러한 것들은 [13,19,43]. dataset (BOP, YCB-V)등 가정내에 object 에서 좋은 성능을 보였다
- 그렇지만 CNN training 을 위해서 엄청나게 많은 수의 데이터가 필요하다 
- 두가지 정도의 접근 방법이 있는데
1) There have been two approaches to create training images for pose estimation methods: rendering synthetic images using textured 3D models

2)  cropping real images and pasting them on random background images
--> 다시 읽기

-- 이 논문의 접근 법은 synthesize images of an object in arbitrary poses using a few cluttered images with pose annotations and a non-textured 3D model of the object 
- 3D models과 color images (16장 이하) 와 color information으로 pose annotations 한 맵만 있으면 된다

## Related work

---
### 이미지 학습을 포함하는 pose estimation
- 21. Kehl, W., Manhardt, F., Tombari, F., Ilic, S., Navab, N.: Ssd-6d: Making rgb based 3d detection and 6d pose estimation great again. In: The IEEE International
Conference on Computer Vision (ICCV) (2017) 1, 3, 9, 12
- 24. Li, Z., Wang, G., Ji, X.: CDPN: Coordinates-based disentangled pose network for
real-time RGB-based 6-DoF object pose estimation. In: Proceedings of the IEEE
International Conference on Computer Vision. pp. 7678–7687 (2019) 1, 2, 3, 10
- 27. Park, K., Patten, T., Vincze, M.: Pix2Pose: Pixel-wise coordinate regression of objects for 6D pose estimation. In: Proceedings of the IEEE International Conference
on Computer Vision. pp. 7668–7677 (2019) 1, 2, 3, 9, 10, 12, 13
- 31. Rad, M., Lepetit, V.: Bb8: A scalable, accurate, robust to partial occlusion method
for predicting the 3d poses of challenging objects without using depth. In: The
IEEE International Conference on Computer Vision (ICCV) (2017) 1, 2, 3, 9, 10
- 34. Sundermeyer, M., Marton, Z.C., Durner, M., Brucker, M., Triebel, R.: Implicit 3d
orientation learning for 6d object detection from rgb images. In: The European
Conference on Computer Vision (ECCV) (2018) 1, 9, 12, 13
- 35. Tekin, B., Sinha, S.N., Fua, P.: Real-time seamless single shot 6d object pose
prediction. In: The IEEE Conference on Computer Vision and Pattern Recognition
(CVPR) (2018) 1, 9, 10
- 45. Zakharov, S., Shugurov, I., Ilic, S.: DPOD: 6D pose object detector and refiner.
In: Proceedings of the IEEE International Conference on Computer Vision. pp.
1941–1950 (2019) 1, 3, 10, 12, 13
---
코드
- 
# 220110
- https://github.com/kirumang/NOL 이거 해보기로 했다
- docker_run.sh를 통해 smot 데이터가 있는 dir를 연결한다 (https://tttsss77.tistory.com/161)
- 2022-01-10 04:28:35.998425: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer.so.6'; dlerror: libnvinfer.so.6: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/lib/x86_64-linux-gnu:/usr/lib/i386-linux-gnu:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
error가 나서 아래 설치를 해줌
- pip install tf-nightly

- 딱히 ubuntu로 빌드할 필요가 없다면,,, 그냥 ananaconda에 올려보자

---
# 새롭게 알게 된 부분
- UV coordinates :  2차원 그림을 3차원 모델로 만드는 3차원 모델링 프로세스( 메시 해체, 텍스처 만들기, 테스처 적용)
